From e91f02df389a81023f7321153dacf7855a2a4c5c Mon Sep 17 00:00:00 2001
From: sebastien baudouin <sebastien.baudouin@windriver.com>
Date: Sat, 25 Apr 2015 08:53:20 +0200
Subject: [PATCH 1/6] Background: Make Graphic working on Qemu machine

This commit makes some modifications to enable QEMU platform support. Those
modification should not be needed in theory but unfortunately are needed if we
want graphic being displayed on screen.
---
 main.cpp | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/main.cpp b/main.cpp
index 42b433c..ad47fb7 100644
--- a/main.cpp
+++ b/main.cpp
@@ -17,6 +17,11 @@
 
 #include <QApplication>
 #include <QQuickView>
+//For QEMU
+#include <QScreen>
+#include <QQuickWindow>
+#include <QQuickItem>
+
 #include <systemd/sd-journal.h>
 
 #define GDP_BACKGROUND_SURFACE_ID 2
@@ -33,6 +38,12 @@ int main(int argc, char *argv[])
     QQuickView view(QUrl(QStringLiteral("qrc:/main.qml")));
 
     view.setProperty("IVI-Surface-ID", GDP_BACKGROUND_SURFACE_ID);
+
+    //For QEMU only
+    view.showFullScreen();
+    view.rootObject()->setWidth(1024);
+    view.rootObject()->setHeight(768);
+
     view.show();
 
     return app.exec();
-- 
1.9.1

